File: app/src/main/java/com/example/daisyling/common/util/download/DownloadListener.java
package com.example.daisyling.common.util.download;

import java.io.File;

/**
 * Created by Emily on 10/11/21
 */
public interface DownloadListener {
    void onFinish(File file);

    void onProgress(int progress, long downloadedLengthKb, long totalLengthKb);

    void onFailed(String errMsg);
}

File: app/src/main/java/com/example/daisyling/common/util/download/DownloadListener.kt
package com.example.daisyling.common.util.download

import java.io.File

/**
 * Created by Emily on 10/11/21
 */
interface DownloadListener {
    fun onFinish(file: File?)
    fun onProgress(progress: Int, downloadedLengthKb: Long, totalLengthKb: Long)
    fun onFailed(errMsg: String?)
}